src/views/profile/bookmarks-view/index.tsx

```diff
@@ -0,0 +1,57 @@
+import { useAllBookmarks } from 'hooks/bookmarks'
+import BaseNewLayout from 'layouts/base-new'
+import AuthenticatedView from 'views/authenticated-view'
+import CardsGridList from 'components/cards-grid-list'
+import Text from 'components/text'
+import Heading from 'components/heading'
+import renderBookmarkCard from './helpers/render-bookmark-cards'
+import s from './bookmarks-view.module.css'
+
+/**
+ * The exported view component that handles wrapping the view content in
+ * `AuthenticatedView`, which automatically handles rendering gated content.
+ */
+const ProfileBookmarksView = () => {
+	return (
+		<AuthenticatedView>
+			<ProfileBookmarksViewContent />
+		</AuthenticatedView>
+	)
+}
+
+/**
+ * The content of the ProfileView that is gated behind authentication.
+ */
+const ProfileBookmarksViewContent = () => {
+	const { bookmarks, isLoading } = useAllBookmarks({ enabled: true })
+
+	if (isLoading) {
+		return null // TODO return loading skeleton
+	}
```

Comment:
Can this be updated to check for first load via `isFetching` and `isRefetching` like [`TutorialCardsGridList`](https://github.com/hashicorp/dev-portal/blob/main/src/components/cards-grid-list/components/tutorial-cards-grid-list/index.tsx#L51-L54) checks? I wonder if we should add an `isFirstLoad` boolean in the return object of these hooks. 💭